src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormatter.cs

```diff
@@ -242,16 +242,28 @@ private TextEdit[] MapEditsToHostDocument(RazorCodeDocument codeDocument, TextEd
             }
         }
 
-        private static bool IsIgnoredByCSharpFormatter(SyntaxNode? parent)
-            => parent?.AncestorsAndSelf().Any(n => n switch
+        private static bool ShouldIgnoreLineCompletely(SyntaxNode? parent, SourceText text)
+        {
+            return parent?.AncestorsAndSelf().Any(n => n switch
             {
                 // C# formatter doesn't touch array initializers
                 InitializerExpressionSyntax { RawKind: (int)CodeAnalysis.CSharp.SyntaxKind.ArrayInitializerExpression } => true,
                 // C# formatter doesn't touch object and collection initializers if they're not empty
                 InitializerExpressionSyntax { Expressions: { Count: > 0 } } => true,
+                // We don't want to format lines that are part of multi-line string literals
+                LiteralExpressionSyntax { RawKind: (int)CodeAnalysis.CSharp.SyntaxKind.StringLiteralExpression } => SpansMultipleLines(n, text),
```

Comment:
Does this mean if we have something like:
```razor
@code
{
var x = @"text
text
text";
}
```

that the `var x` at the beginning will no longer be indented when formatting?

Reply:
No, because this is looking up the tree. The `var` tokens parent would be a LocalDeclarationSyntax, so it won't fall into any of these cases.

This kicks in on the second line, where the `test` tokens parent is a string literal.

If this was descending then yes, it would be wrong for the `var x` line.